app/controllers/data_sets_controller.rb

```diff
@@ -305,16 +327,59 @@ def dataFileUpload
       end
     end
 
-    if params[:file]
-      @filename = DataSet.get_next_name(project, params[:file].original_filename.split('.')[0])
-    else
-      @filename = DataSet.get_next_name(project, 'Google docs')
+    if params[:file] && params[:file].original_filename.split('.')[1] == 'zip'
+      Zip::File.open(params[:file].path) do |zip_file|
+
+        results = []
+        filenames = []
+        zip_file.each do |entry|
+          if entry.directory?
+            next
+          elsif entry.file?
+            if entry.name.include? '__MACOSX' or entry.name.include? 'DS_Store'
+              next
+            end
+            begin
+              tempfile = Tempfile.new(entry.name.split('/')[1])
+              tempfile.write(entry.get_input_stream.read)
+
+              uploader = FileUploader.new
+              data_obj = uploader.generateObject(tempfile)
+              results.push(uploader.match_headers(project, data_obj))
+
+              filenames.push(DataSet.get_next_name(project, entry.name.split('/')[1].split('.')[0]))
+
+            rescue Exception
+              flash[:error] = "Error reading file #{entry.name}"
+              next
```

Comment:
Shouldn't this rescue handle the cause of `__MACOSX` and `DS_Store` files?  There seems to be no reason to special case those.